src/main/webapp/app/exercises/shared/feedback/item/feedback-item.ts

```diff
@@ -11,4 +12,17 @@ export class FeedbackItem implements FeedbackNode {
     text?: string; // this is typically feedback.detailText
     positive?: boolean;
     color?: FeedbackColor;
+    feedbackReference: FeedbackReference;
+}
+
+export class FeedbackReference {
+    feedbackId: number;
```

Comment:
Cant we just keep the entire `feedback`, `result` as reference instead of only the ID?

Reply:
Yes, we can. Adapted it.
The original intention was to keep the stored information in `feedbackItem` to a minimum with only non-`undefined` values. I guess the IDs should always be present anyway and an `undefined` `hasLongFeedbackText` can safely be assumed to be `false`.